syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -308,20 +346,134 @@ def __truediv__(self, shares: dict, divisor):
         for location, pointer in shares.items():
             divided_shares[location] = pointer / divisor
 
-        return AdditiveSharingTensor(
-            field=self.field, crypto_provider=self.crypto_provider
-        ).set_shares(divided_shares)
+        return divided_shares
 
     @overloaded.method
     def mod(self, shares: dict, modulus: int):
         assert isinstance(modulus, int)
+
         moded_shares = {}
         for location, pointer in shares.items():
             moded_shares[location] = pointer % modulus
 
-        return AdditiveSharingTensor(
-            field=self.field, crypto_provider=self.crypto_provider
-        ).set_shares(moded_shares)
+        return moded_shares
 
     def __mod__(self, *args, **kwargs):
         return self.mod(*args, **kwargs)
+
+    ## SECTION SNN
+
+    def relu(self):
+        return securenn.relu(self)
+
+    def positive(self):
+        # self >= 0
+        return securenn.relu_deriv(self)
+
+    def __gt__(self, other):
+        r = self - other - 1
+        print(r)
+        return r.positive()
+
+    def gt(self, other):
+        return self > other
+
+    def __ge__(self, other):
+        return (self - other).positive()
+
+    def ge(self, other):
+        return self >= other
+
+    def __lt__(self, other):
+        return (other - self - 1).positive()
+
+    def lt(self, other):
+        return self < other
+
+    def __le__(self, other):
+        return (other - self).positive()
+
+    def le(self, other):
+        return self <= other
+
+    def __eq__(self, other):
```

Comment:
Could we set this up so that the logic lives in the not __ option. That is more standard

Reply:
Not sure, as in practice we don't use the non __ ones

Reply:
the way that pytorch does it is the other way ie __eq__(self, other) = eq(self,other)